src/main/java/com/cloudogu/smeagol/DevelopmentDispatcher.java

```diff
@@ -0,0 +1,99 @@
+package com.cloudogu.smeagol;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.ByteStreams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The development dispatcher proxies every non api request to configured {@code ui.url}.
+ */
+public class DevelopmentDispatcher implements Dispatcher {
+
+    private String target;
+
+    @Autowired
+    public DevelopmentDispatcher(@Value("${ui.url}") String target) {
+        this.target = target;
+    }
+
+    @Override
+    public boolean needsToBeDispatched(String uri) {
+        return !(uri.startsWith("/api") || uri.startsWith("api"));
+    }
+
+    @Override
+    public void dispatch(HttpServletRequest request, HttpServletResponse response, String uri) throws IOException {
+        URL url = createProxyUrl(uri);
+
+        HttpURLConnection connection = openConnection(url);
+        connection.setRequestMethod(request.getMethod());
+        copyRequestHeaders(request, connection);
+
+        if (request.getContentLengthLong() > 0) {
+            copyRequestBody(request, connection);
+        }
+
+        int responseCode = connection.getResponseCode();
+        response.setStatus(responseCode);
+        copyResponseHeaders(response, connection);
+        copyResponseBody(response, connection);
+    }
+
+    private void copyResponseBody(HttpServletResponse response, HttpURLConnection connection) throws IOException {
+        try (InputStream input = connection.getInputStream(); OutputStream output = response.getOutputStream()) {
+            ByteStreams.copy(input, output);
+        }
+    }
+
+    private void copyResponseHeaders(HttpServletResponse response, HttpURLConnection connection) {
+        Map<String, List<String>> headerFields = connection.getHeaderFields();
+        for (Map.Entry<String,List<String>> entry : headerFields.entrySet()) {
+            if (entry.getKey() != null && ! entry.getKey().equalsIgnoreCase("Transfer-Encoding")) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the "Transfer-Encoding" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1132)